libraries/chain/controller.cpp

```diff
@@ -337,7 +337,12 @@ struct controller_impl {
       set_activation_handler<builtin_protocol_feature_t::crypto_primitives>();
 
       self.irreversible_block.connect([this](const block_state_ptr& bsp) {
-         get_wasm_interface().current_lib(bsp->block_num);
+         EOS_ASSERT( appbase::app().executor().is_write_window(), misc_exception, "write window expected for irreversible_block signal"); // read-only threads are idle, threaded_wasmifs are not used
```

Comment:
If signals are called from read-window all kinds of assumptions do not hold. You could move this assert into `producer_plugin`'s handling of `irreversible_block signal`.

Reply:
Will move this to producer plugin.